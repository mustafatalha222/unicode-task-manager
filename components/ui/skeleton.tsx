import { cn } from '@/lib/utils'
import { useTranslations } from 'next-intl'

function Skeleton({ className, ...props }: React.HTMLAttributes<HTMLDivElement>) {
  return <div className={cn('animate-pulse rounded-md bg-primary/10', className)} {...props} />
}

function CustomSkeleton({
  className,
  loading = false,
  length = 0,
  ...props
}: React.HTMLAttributes<HTMLDivElement> & { loading: boolean; length: number }) {
  const t = useTranslations()
  return (
    <>
      {loading && !length ? (
        <div className={cn('animate-pulse rounded-md bg-primary/10', className)} {...props} />
      ) : (
        !length && <h4 className="text-xl font-bold flex justify-center">{t('No Data Found')}</h4>
      )}
    </>
  )
}

export { Skeleton, CustomSkeleton }
