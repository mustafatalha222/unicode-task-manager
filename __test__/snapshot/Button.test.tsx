import { Button } from '@/components/ui/button'
import renderer from 'react-test-renderer'

it('renderer language Switcher unchange', () => {
  const tree = renderer.create(<Button variant={'destructive'}>Button Snapshot</Button>).toJSON()
  expect(tree).toMatchSnapshot()
})
