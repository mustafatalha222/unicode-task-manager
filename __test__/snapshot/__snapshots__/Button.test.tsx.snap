// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer language Switcher unchange 1`] = `
<button
  className="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-destructive text-destructive-foreground shadow-sm hover:bg-destructive/90 h-9 px-4 py-2"
>
  Button Snapshot
</button>
`;
